js/common/lib/tensor-impl.ts

```diff
@@ -102,7 +102,8 @@ export class Tensor implements TensorInterface {
           break;
         }
         case 'gpu-buffer': {
-          if (type !== 'float32' && type !== 'int32') {
+          if ((type !== 'float32' && type !== 'float16' && type !== 'int32' && type !== 'int64' && type !== 'uint32' &&
```

Comment:
Can we use something like `type is typeof GpuBufferDataTypea` to avoid hardcoded values?

Reply:
I like the idea of `type is typeof GpuBufferDataType` but seems typescript does not allow doing similar things...